### PR TITLE
GEODE-3563: use a timeout for newly created sockets in TcpConduit.run()

### PR DESCRIPTION
**Pull request by galen-pivotal**
Also close newly accepted sockets in TcpConduit.run() if SSL configuration fails (or any other IOException).

I'm not sure the best way to test the `run()` method itself, so I haven't attempted it here. l'm not sure what the best way to do that is and leaks are hard to test in general. Let me know if you have ideas.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java [hunk 1]
```diff
@@ -968,8 +968,11 @@ public Socket connect(InetAddress inetadd, int port, int timeout,
 
   /**
    * Will be a server socket... this one simply registers the listeners.
+   *
+   * @param timeout the socket's timeout will be set to this.
```
Comment (PivotalSarge) [inline geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java]: Which units?

File: geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java [hunk 2]
```diff
@@ -990,6 +993,7 @@ public void configureServerSSLSocket(Socket socket) throws IOException {
               ex);
           throw ex;
         }
+        // else ignore
```
Comment (PivotalSarge) [inline geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java]: Does this refer to a "live" if-statement?

Comment (galen-pivotal) [reply geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java]: I'm not sure what you mean by "live". It refers to the if before it. I'm not sure that's what we really should want to do, but it's the current behaviour.

File: geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java (hunk 1 above)
Comment (galen-pivotal) [reply geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java]: milliseconds. Changed.

File: geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java [hunk 3]
```diff
@@ -968,8 +968,11 @@ public Socket connect(InetAddress inetadd, int port, int timeout,
 
   /**
    * Will be a server socket... this one simply registers the listeners.
+   *
+   * @param timeout the socket's timeout will be set to this (in milliseconds).
    */
-  public void configureServerSSLSocket(Socket socket) throws IOException {
+  public void configureServerSSLSocket(Socket socket, int timeout) throws IOException {
+    socket.setSoTimeout(timeout);
     if (socket instanceof SSLSocket) {
```
Comment (WireBaron) [inline geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java]: This seems like the wrong failure behavior (silently continue) if we call this on a non-SSL socket.

Comment (galen-pivotal) [reply geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java]: It seems like it's mostly the name of this method that is odd. It looks like the server socket gets configured for SSL when it's created, and child sockets will be SSL or not based on what the server socket is.

Perhaps it should be `handshakeIfSocketIsSSL` or something?

**Pull request merged**